zerver/actions/realm_settings.py

```diff
@@ -475,6 +476,17 @@ def do_change_realm_plan_type(
         # We do not allow public access on limited plans.
         do_set_realm_property(realm, "enable_spectator_access", False, acting_user=acting_user)
 
+    if old_value in [Realm.PLAN_TYPE_PLUS, Realm.PLAN_TYPE_SELF_HOSTED] and plan_type not in [
+        Realm.PLAN_TYPE_PLUS,
+        Realm.PLAN_TYPE_SELF_HOSTED,
```

Comment:
We would not disable this feature in self-hosted realms, but I am not sure whether self-hosted orgs will use this code to downgrade the plan so we can probably remove that part.